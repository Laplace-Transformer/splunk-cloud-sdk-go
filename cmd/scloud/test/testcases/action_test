action create-action-email --name scloud_test_action_test --members kk@splunk.com --members aa@splunk.com
action update-action-email-mutable --action-name scloud_test_action_test --subject "newsubject"
action get-action --action-name scloud_test_action_test

action get-public-webhook-keys

action create-action-webhook --name scloud_test_action_test_wh --title newtitle --webhook-payload "{{ .name }} is a {{ .species }}" --webhook-url "https://webhook.site/test"
action update-action-webhook-mutable --action-name scloud_test_action_test_wh --title "newtitle"

action trigger-action --action-name scloud_test_action_test_wh
action get-action-status --action-name scloud_test_action_test --status-id CN4uIrr8qsmBVzkhlg0niUs39euOr6Jb

action  list-actions

action delete-action --action-name scloud_test_action_test
action delete-action --action-name scloud_test_action_test_wh
