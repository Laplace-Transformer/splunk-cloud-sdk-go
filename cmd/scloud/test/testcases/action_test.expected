#testcase: action create-action-email --name scloud_test_action_test --members kk@splunk.com --members aa@splunk.com
REQUEST URL:action/v1beta2/actions
REQUEST BODY:{{"kind":"email","name":"scloud_test_action_test","members":["kk@splunk.com","aa@splunk.com"]}}

#testcase: action update-action-email-mutable --action-name scloud_test_action_test --subject "newsubject"
REQUEST URL:action/v1beta2/actions/scloud_test_action_test
REQUEST BODY:{{"subject":"newsubject"}}

#testcase: action get-action --action-name scloud_test_action_test
REQUEST URL:action/v1beta2/actions/scloud_test_action_test
REQUEST BODY:<nil>

#testcase: action get-public-webhook-keys
REQUEST URL:action/v1beta2/webhook/keys
REQUEST BODY:<nil>

#testcase: action create-action-webhook --name scloud_test_action_test_wh --title newtitle --webhook-payload "{{ .name }} is a {{ .species }}" --webhook-url "https://webhook.site/test"
REQUEST URL:action/v1beta2/actions
REQUEST BODY:{{"kind":"webhook","name":"scloud_test_action_test_wh","webhookPayload":"{{ .name }} is a {{ .species }}","webhookUrl":"https://webhook.site/test","title":"newtitle"}}

#testcase: action update-action-webhook-mutable --action-name scloud_test_action_test_wh --title "newtitle"
REQUEST URL:action/v1beta2/actions/scloud_test_action_test_wh
REQUEST BODY:{{"title":"newtitle"}}

#testcase: action trigger-action --action-name scloud_test_action_test_wh
REQUEST URL:action/v1beta2/actions/scloud_test_action_test_wh
REQUEST BODY:{{"actionMetadata":{}}}

#testcase: action get-action-status --action-name scloud_test_action_test --status-id CN4uIrr8qsmBVzkhlg0niUs39euOr6Jb
REQUEST URL:action/v1beta2/actions/scloud_test_action_test/status/CN4uIrr8qsmBVzkhlg0niUs39euOr6Jb
REQUEST BODY:<nil>

#testcase: action  list-actions
REQUEST URL:action/v1beta2/actions
REQUEST BODY:<nil>

#testcase: action delete-action --action-name scloud_test_action_test
REQUEST URL:action/v1beta2/actions/scloud_test_action_test
REQUEST BODY:<nil>

#testcase: action delete-action --action-name scloud_test_action_test_wh
REQUEST URL:action/v1beta2/actions/scloud_test_action_test_wh
REQUEST BODY:<nil>

